driver/userspace.c

```diff
@@ -324,17 +323,18 @@ WnbdCreateConnection(PWNBD_EXTENSION DeviceExtension,
         return STATUS_SHUTDOWN_IN_PROGRESS;
     }
 
+    PWNBD_DISK_DEVICE Device = (PWNBD_DISK_DEVICE) Malloc(sizeof(WNBD_DISK_DEVICE));
```

Comment:
no need to move this allocation here, you can just set Device to NULL so that you may use it in the ``Exit`` routine.